src/metadata.rs

```diff
@@ -224,6 +225,13 @@ impl Metadata {
         })
     }
 
+    pub fn from_raw(src: *const grpc_metadata_array) -> Self {
```

Comment:
I'm confused about how to do this. I tried it this way, and it gave me a segmentation fault:
`            let m = src as *const metadata;
            return metadata::from_raw_parts((*m).0.metadata, (*m).0.count, (*m).0.capacity);
`
Can you give an example of how this could work without cloning?

Reply:
Like this:
```
let mut res = Metadata::with_capacity(0);
grpcio_sys::grpcwrap_batch_context_move_recv_initial_metadata(ctx, &mut res as *mut _);
```

The implementation of `grpcwrap_batch_context_move_recv_initial_metadata` is:
```
GPR_EXPORT void GPR_CALLTYPE
grpcwrap_batch_context_take_recv_initial_metadata(
    const grpcwrap_batch_context* ctx, grpc_metadata_array* res) {
    grpcwrap_metadata_array_move(res, &(ctx->recv_initial_metadata));
}
```

The key point is use a pointer, which point to the rust stack to receive the moved content from C side. What you write will leave two pointer points to the same metadata, the first one is src->metadata, the second is the returned value. So it will double free eventually. `grpcwrap_metadata_array_move` will handle the pointers correctly.

Reply:
I attempted to implement this, but it is still segfaulting. Can you take a look at the head of my branch and see if you can fix the segfault?

Reply:
Can you also provide the source code?